### PR TITLE
Add partition filters Testing to all Formats

### PR DESCRIPTION
**Pull request by vamshigv**
## *Important Read*
- *Please ensure the GitHub issue is mentioned at the beginning of the PR*

## What is the purpose of the pull request

Add tests for testing partition filter behavior

## Brief change log

1. Add tests for partition filters work for all formats.
2. Basic filters for Iceberg & Delta while more advanced filters for Hudi

## Verify this pull request

This change added tests and can be verified as follows:

### CODE REVIEW

File: core/src/main/java/io/onetable/hudi/HudiTableManager.java [hunk 1]
```diff
@@ -99,6 +99,7 @@ HoodieTableMetaClient initializeHudiTable(String tableDataPath, OneTable table)
     try {
       return HoodieTableMetaClient.withPropertyBuilder()
           .setCommitTimezone(HoodieTimelineTimeZone.UTC)
+          .setHiveStylePartitioningEnable(true)
```
[the remark this answers is not in the document]
Comment (the-other-tim-brown) [reply core/src/main/java/io/onetable/hudi/HudiTableManager.java:102]: You should use the `DataLayoutStrategy` of the table

**Comment by vamshigv** [comment]
I dug a bit more and commented at relevant places. Both iceberg and Delta uses hive partitioning by default and reflected that in DataLayoutStrategy.

File: core/src/main/java/io/onetable/hudi/HudiTableManager.java [hunk 2]
```diff
@@ -96,9 +97,12 @@ HoodieTableMetaClient initializeHudiTable(String tableDataPath, OneTable table)
     keyGeneratorClass =
         getKeyGeneratorClass(
             table.getPartitioningFields(), table.getReadSchema().getRecordKeyFields());
+    boolean hiveStylePartitioningEnabled =
```
Comment (the-other-tim-brown) [inline core/src/main/java/io/onetable/hudi/HudiTableManager.java:100]: Can you update the tests to validate this?

File: core/src/test/java/io/onetable/TestSparkDeltaTable.java [hunk 3]
```diff
@@ -101,7 +101,7 @@ public List<Row> insertRows(int numRows) {
   }
 
   public List<Row> insertRowsForPartition(int numRows, Integer partitionValue) {
-    return insertRowsForPartition(numRows, partitionValue, partitionField);
```
Comment (the-other-tim-brown) [inline core/src/test/java/io/onetable/TestSparkDeltaTable.java:104]: This is changing from a field to a value, is that intentional?

File: core/src/main/java/io/onetable/hudi/HudiTableManager.java (hunk 2 above)
Comment (vamshigv) [reply core/src/main/java/io/onetable/hudi/HudiTableManager.java:100]: yeah did that.

File: core/src/test/java/io/onetable/TestSparkDeltaTable.java (hunk 3 above)
Comment (vamshigv) [reply core/src/test/java/io/onetable/TestSparkDeltaTable.java:104]: yes previously it was in error state, we need partition values in method signature.

**Review by the-other-tim-brown — APPROVED** [review]

**Pull request merged**